Finished Implementation of Class Based Structure.

GruescuDanziel opened the pull request:
Going to add functionality for meetingRoomRepository after talk with Azure Admin.

GruescuDanziel committed:
Finished Implementation of Class Based Structure.
Going to add funcitonality for meetings after talk with admin.

daniel-mita commented on src/dto/msgraph/meetingRoomDtos/create.meetingRoom.dto.ts:
```diff
@@ -0,0 +1,7 @@
+    meetingRoomName: string;
+    meetingRoomDescription: string;
+    meetingRoomCapacity: number;
+    meetingRoomeEail: string;
```
Nu trebuie sa pui meeting(sau numele obiectului) in fata fiecarei proprietati, de exemplu 'roomName' in loc de 'meetingRoomName'.

CatalinTiu replied:
Am schimbat naming convention peste tot unde trebuia simplificat

daniel-mita commented on src/dto/msgraph/meetingRoomDtos/delete.meetingRoom.dto.ts:
```diff
@@ -0,0 +1,5 @@
+export interface DeleteMeetingRoomDto {
+    meetingRoomName: string;
+    meetingRoomEmail: string;
```
Si aici

daniel-mita commented on src/dto/msgraph/eventDtos/create.event.dto.ts:
```diff
@@ -0,0 +1,7 @@
+export interface CreateEventDto {
```
Aici la fel

daniel-mita commented on src/repository/msgraph/meetingRoom.repository.ts:
```diff
@@ -0,0 +1,33 @@
+    try {
+      const meetingRooms = await axios.get(
+        "https://graph.microsoft.com/v1.0/users",
+        { headers: { Authorization: process.env.AUTH_KEY } }
```
Pentru env variables iti sugerez sa creezi un fisier numit example.env. Aici poti tine placeholdere pentru toate variabilele de care ai nevoie in asa fel in cat toti o sa stie mai apoi de ce au nevoie pentru configurare.
Ex: 'AUTH_KEY=Your auth key from microsoft'

CatalinTiu replied:
am creat un fisier example.env unde am pus un example

daniel-mita commented on src/repository/msgraph/meetingRoom.repository.ts:
```diff
@@ -0,0 +1,33 @@
+  async getAllMeetingRooms() {
+    try {
+      const meetingRooms = await axios.get(
+        "https://graph.microsoft.com/v1.0/users",
```
Incearca sa tii tot ce e legat de url's sau string uri hard-coded in constante

CatalinTiu replied:
am creat un folder si fisier de constants unde am definit call-urile de api si apoi le exportam in restul proiectului unde facem GET-uri si POST-uri

daniel-mita commented on src/routes/Msgraph.routes.ts:
```diff
@@ -0,0 +1,43 @@
+//Events
+//GET - Get All Events
+//POST - Create An Event
+//==============================
```
Foarte tare ideea cu documentatul rutelor, poate ulterior mutat si dupa actualizat intr un md file gen readMe

apcraiovan commented on src/models/msgraph/event.entity.ts:
```diff
@@ -0,0 +1,22 @@
+export interface Event {
```
daca nu ai o clasa care sa fie salvata in DB - nu se numeste entity...eventual e model sau si mai bine cred ca e dto.
Daca trebuie salvata in DB , atunci nu e interfate ci poti defini o clasa Event de exemplu 
`@Table
  class Event extends Model<Event> {
  ...
  }
`
:)

CatalinTiu replied:
Am mutat tot din entity in dto folder

apcraiovan commented on src/repository/msgraph/event.repository.ts:
```diff
@@ -0,0 +1,30 @@
+import axios from "axios";
```
call-urile externe le faci in service class, nu repository :) .
Repository se refera doar la interogarea Db-ului si manipularea entitatiilor.

CatalinTiu replied:
am mutat call-urile externe din repository in service files

apcraiovan commented on src/repository/msgraph/participant.repository.ts:
```diff
@@ -0,0 +1,13 @@
+import axios from "axios";
```
nu cred ca ai nevoie de repo si entity daca nu faci nici o operatie in DB :)

CatalinTiu replied:
am sters cele 2 foldere de ms grpah entity si ms graph repo si mutat fisierele la locul lor

apcraiovan commented on src/service/msgraph/meetingRoom.service.ts:
```diff
@@ -0,0 +1,42 @@
+import { MeetingRoom } from "../../models/msgraph/meetingRoom.entity";
```
adauga logica in serviciu din repo, te rog.

apcraiovan reviewed (changes requested):
mai putin :)

CatalinTiu committed:
made some changes

GruescuDanziel committed:
Made the modifications

apcraiovan commented on src/constants/constants.ts:
```diff
@@ -0,0 +1,6 @@
+const CONSTANTS = {
```
ce nume original :D . stim ca sunt constante asa ca numite cumva :) exemplu:
`export MS_GRAPH_PATH` si apoi aveti  MS_GRAPH_PATH.USERS sau EVENTS.

CatalinTiu replied:
schimbat nume la constante

apcraiovan commented on src/routes/Msgraph.routes.ts:
```diff
@@ -0,0 +1,43 @@
+import { Router } from "express";
```
redenumiti fisierul sa fie cu litera m mic :)

CatalinTiu replied:
redenumit fisier

apcraiovan commented on src/service/msgraph/events.service.ts:
```diff
@@ -0,0 +1,33 @@
+    try {
+      const events = await axios.get(
+        CONSTANTS.MS_GRAPH_PATH_EVENTS,
+        { headers: { authorization: process.env.AUTH_KEY } }
```
nu e gresit dar ca si best practice...as vrea sa creati un obiect chiar aici in service...care sa contina partea de authorization...si astfel nu mai aveti cod redundant aici si la linia 25. :)

CatalinTiu replied:
creat constructor pentru header

apcraiovan commented on src/service/msgraph/meetingRoom.service.ts:
```diff
@@ -0,0 +1,53 @@
+    try {
+      const meetingRooms = await axios.get(
+        CONSTANTS.MS_GRAPH_PATH_EVENTS,
+        { headers: { Authorization: process.env.AUTH_KEY } }
```
la fel ca mai sus.

apcraiovan commented on src/service/msgraph/meetingRoom.service.ts:
```diff
@@ -0,0 +1,53 @@
+  async deleteMeetingRoom(
+    deleteMeetingRoomDto: DeleteMeetingRoomDto
+  ): Promise<void> {
+    // const { meetingRoomName, meetingRoomEmail } = deleteMeetingRoomDto;
```
commenturi nefolositoare...daca nu va trebuie stergeti metoda...daca chiar mai aveti ceva de facut la ea...cand puneti comment incercati sa folositi un TODO ...etc.

CatalinTiu replied:
sters comentarii sau inlocuit cu TODO unde era relevant

apcraiovan reviewed (changes requested):
mai un pic. :)

CatalinTiu committed:
made modifications according to comments

GruescuDanziel committed:
Added the bearer token renewal

GruescuDanziel committed:
Commit the deleteRoomDTO

CatalinTiu committed:
updated headers.config and example.env

CatalinTiu committed:
added getParticipant

GruescuDanziel committed:
Updates on meetingRoom Service

GruescuDanziel committed:
Updated Participants For groups

GruescuDanziel committed:
Changes by catalin

GruescuDanziel committed:
Updated Participants For groups

bogdanhotea-doctari reviewed (commented):
Dani trebe sa pui .idea in gitignore-ul din root-ul proiectului :)

CatalinTiu committed:
solved getToken error and .gitignore

GruescuDanziel committed:
Delete .idea directory

GruescuDanziel committed:
Merge branch 'ClassBassedOutlookApi' of github.com:apcraiovan/meeting-rooms-booking-api into ClassBassedOutlookApi

GruescuDanziel committed:
Finished the outlook api implementation

GruescuDanziel committed:
Almost done

GruescuDanziel committed:
Finsihed implementation for the first version of FE

GruescuDanziel commented:
Features added:

- Added the main endpoint /msgraph that is used to access the msgraph API
- Added an endpoint for each category of request (Event, User, Group)
  - The event category uses GET and POST
  - The user category uses GET
  - The group category uses GET
- Created the msgraph.utils to gather the token with data from .env
  - The value of the variables in .env is taken from the Registered App Dashboard
  - The main function "getAccessToken" is used to request the token and to send it forward to "getHeaderConfig"
  - The function "getHeaderConfig" is used to generate the headers for each request
